### PR TITLE
Test and refactor `GetDatasetDetails`

### PR DESCRIPTION
**Pull request by kavir1698**

### CODE REVIEW

File: datasetUtils/getDatasetDetails.go [hunk 1]
```diff
@@ -32,55 +46,67 @@ func GetDatasetDetails(client *http.Client, APIServer string, accessToken string
 		if end > len(datasetList) {
 			end = len(datasetList)
 		}
-
-		var filter = `{"where":{"pid":{"inq":["` +
-			strings.Join(datasetList[i:end], `","`) +
-			`"]}},"fields":{"pid":true,"sourceFolder":true,"size":true,"ownerGroup":true}}`
-
+
```
Comment (minottic) [inline datasetUtils/getDatasetDetails.go:49]: extra tab

File: datasetUtils/getDatasetDetails.go [hunk 2]
```diff
@@ -32,55 +46,67 @@ func GetDatasetDetails(client *http.Client, APIServer string, accessToken string
...
-
-		var filter = `{"where":{"pid":{"inq":["` +
-			strings.Join(datasetList[i:end], `","`) +
-			`"]}},"fields":{"pid":true,"sourceFolder":true,"size":true,"ownerGroup":true}}`
-
+		
+		filter := `{"where":{"pid":{"inq":["` +
+		strings.Join(datasetList[i:end], `","`) +
+		`"]}},"fields":{"pid":true,"sourceFolder":true,"size":true,"ownerGroup":true}}`
+
```
Comment (minottic) [inline datasetUtils/getDatasetDetails.go:53]: extra tab. Maybe we should review the linter option

File: datasetUtils/getDatasetDetails_test.go
Comment (minottic) [inline datasetUtils/getDatasetDetails_test.go:1]: can we add one test where it finds a dataset (by mocking the server to return the id)?

File: datasetUtils/getDatasetDetails_test.go [hunk 3]
```diff
@@ -0,0 +1,120 @@
...
+	// Check the result
+	if len(datasets) != 1 {
+		t.Errorf("Expected 1 dataset, got %d", len(datasets))
+	} else {
+		dataset := datasets[0]
+		if dataset.Pid != "123" || dataset.SourceFolder != "/path/to/dataset" || dataset.Size != 1024 || dataset.OwnerGroup != "group1" || dataset.NumberOfFiles != 10 {
+			t.Errorf("Dataset details do not match expected values")
+		}
+	}
+}
```
Comment (minottic) [inline datasetUtils/getDatasetDetails_test.go:120]: missing new line

**Review by sbliven — APPROVED** [review]

**Pull request merged**